src/api.v2/model/ExperimentExecution.js

```diff
@@ -56,6 +57,21 @@ class ExperimentExecution extends BasicModel {
 
     await sql(tableNames.EXPERIMENT_EXECUTION).insert(copyRows);
   }
+
+  async updateExecution(experimentId, pipelineType, newExecution, retryParams) {
```

Comment:
Is it worth it to create this function? 

just calling

```
await this.upsert(
      {experiment_id: experimentId, pipeline_type: pipelineType,},
      { ...newExecution, retry_params: retryParams},
    );
```

seems fairly simple, or is this being called in more than one place? Perhaps in that case I can see the us

Reply:
It is called in every execution of a pipeline (not just when retrying). It adds a layer of abstraction, but I don't think readibility suffers from it